src/util/string_expr.h

```diff
@@ -14,37 +14,28 @@ Author: Romain Brenguier, romain.brenguier@diffblue.com
 
 #include <util/std_expr.h>
 #include <util/arith_tools.h>
+#include <util/refined_string_type.h>
 
-class string_exprt: public struct_exprt
+// Given an representation of strings as exprt that implements `length` and
+// `content` this provides additional useful methods.
+template<typename child_t>
```

Comment:
@reuk or @LAJW could perhaps opine on these template shenanigans

Reply:
Yes, I'm not sure I see the need for this at all. You could write this as a set of free template functions which take some type `T` as an argument, and document that these functions require `length` and `content` to be implemented. It's not necessary to have these be member functions (as far as I can tell) because they don't access private data or functions, and we're not doing virtual dispatch.

In general, lower coupling is better, and inheritance is one of the strongest forms of coupling.

Reply:
I think it's a suitable intermediary step - it preserves the interface of the class, while at the same time letting remove code duplication between two similar child classes.

Best approach here would be to use inheritance with virtual member functions. Unfortunately classes inheriting from `exprt` cannot have any data members. Pointer to the virtual table is a member, therefore this approach could not be used (it has been tried before and caused this version to be written).